src/blockchain/txpool.ts

```diff
@@ -88,6 +89,20 @@ export class TxPool {
       // trying to make a SecondaryPlainPreDigest
       const newSlot = compactAddLength(u8aConcat('0x0200000000', bnToU8a(expectedSlot, { isLe: true, bitLength: 64 })))
       newLogs = [{ PreRuntime: [consensus.consensusEngine, newSlot] }, ...consensus.rest]
+    } else if (consensus?.consensusEngine?.toString() == 'nmbs') {
+      const nmbsKey = stringToHex('nmbs')
+      newLogs = [
+        {
+          // Using previous block author
+          PreRuntime: [
+            consensus.consensusEngine,
+            parentHeader.digest.logs
+              .find((log) => log.isPreRuntime && log.asPreRuntime[0].toHex() == nmbsKey)
+              ?.asPreRuntime[1].toHex(),
+          ],
+        },
+        ...consensus.rest,
+      ]
```

Comment:
please put this line here until we find better solution 
```suggestion
          head.pushStorageLayer().set(compactHex(meta.query.randomness.notFirstBlock()), StorageValueKind.Deleted)
      ]
```